src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -19,18 +21,21 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import jenkins.tasks.SimpleBuildStep;
 import net.sf.json.JSONObject;
 
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.StaplerRequest;
 
-public class LogParserPublisher extends Recorder implements Serializable {
+public class LogParserPublisher extends Recorder implements SimpleBuildStep, Serializable {
     private static final long serialVersionUID = 1L;
-    public final boolean unstableOnWarning;
-    public final boolean failBuildOnError;
-    public final boolean showGraphs;
-    public final String parsingRulesPath;
-    public final boolean useProjectRule;
-    public final String projectRulePath;
+    public boolean unstableOnWarning;
+    public boolean failBuildOnError;
+    public boolean showGraphs;
+    public String parsingRulesPath;

```

Comment:
@tfennelly 
The problem is that setters are being exposed as a public API => they can be used from a random place now. Jenkins processes are concurrent by default, hence the code may randomly blow up if a user starts using setters in his plugins.

Since you're adding new fields, you cannot even restrict the usage of setters, because users won't be able to configure data structures in their plugins and unit tests.